src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -83,26 +118,41 @@ const CharacterForm = ({ onClose }) => {
     species,
     numberOfFilms,
   }) => {
+    const commonAttributes = {
+      name,
+      eyeColor,
+      height,
+      gender: gender?.value,
+      birthYear,
+      filmConnection: {
+        totalCount: numberOfFilms,
+      },
+    }
+
     dispatch(
-      addCharacter({
-        id: nanoid(),
-        name,
-        eyeColor,
-        height,
-        gender: gender?.value,
-        birthYear,
-        homeworld: {
-          id: nanoid(),
-          name: homeworld,
-        },
-        species: {
-          id: nanoid(),
-          name: species?.value,
-        },
-        filmConnection: {
-          totalCount: numberOfFilms,
-        },
-      })
+      isEdit
+        ? editCharacter({
+            id: character.id,
+            homeworld: {
+              name: homeworld,
```

Comment:
Please create a function that will prepare your character data given the form data as input